libs/muddle/src/kademlia/peer_tracker.cpp

```diff
@@ -653,26 +682,28 @@ void PeerTracker::ConnectToDesiredPeers()
         best_peer = closest_peers[idx].address;
 
         // Skipping own address
-        if (best_peer == own_address_)
+        if (best_peer == my_address)
         {
           ++idx;
           continue;
         }
 
-        auto it = last_pull_from_peer_.find(best_peer);
-
-        // Breaking if we never did a pull before
-        if (it == last_pull_from_peer_.end())
         {
-          break;
-        }
+          FETCH_LOCK(pull_mutex_);
+          auto it = last_pull_from_peer_.find(best_peer);
 
-        // Breaking if it is sufficiently long ago
-        if (ToSeconds(Clock::now() - it->second) > 300)  // TODO(tfr): Fix parameter
-        {
-          break;
-        }
+          // Breaking if we never did a pull before
+          if (it == last_pull_from_peer_.end())
+          {
+            break;
+          }
 
+          // Breaking if it is sufficiently long ago
+          if (ToSeconds(Clock::now() - it->second) > 300)  // TODO(tfr): Fix parameter
```

Comment:
NP: `Clock::now()`  should probably be evaluated once at loop's start, should be enough for one second-resolution.